core-api/src/test/java/com/optimizely/ab/internal/DefaultLRUCacheTest.java

```diff
@@ -0,0 +1,171 @@
+        assertEquals("user1", cache.linkedList.get(2).key);
+
+        assertEquals(Arrays.asList("segment1", "segment2"), cache.lookup("user1"));
+
+        // Lookup should move user1 to top of the list and push down others.
+        assertEquals("user1", cache.linkedList.get(0).key);
+        assertEquals("user3", cache.linkedList.get(1).key);
+        assertEquals("user2", cache.linkedList.get(2).key);
+
+        assertEquals(Arrays.asList("segment3", "segment4"), cache.lookup("user2"));
+
+        // Lookup should move user2 to top of the list and push down others.
+        assertEquals("user2", cache.linkedList.get(0).key);
+        assertEquals("user1", cache.linkedList.get(1).key);
+        assertEquals("user3", cache.linkedList.get(2).key);
+
+        assertEquals(Arrays.asList("segment5", "segment6"), cache.lookup("user3"));
+
+        // Lookup should move user3 to top of the list and push down others.
+        assertEquals("user3", cache.linkedList.get(0).key);
+        assertEquals("user2", cache.linkedList.get(1).key);
+        assertEquals("user1", cache.linkedList.get(2).key);
+    }
+
+    @Test
+    public void whenCacheIsDisabled() {
+        DefaultLRUCache<List<String>> cache = new DefaultLRUCache<>();
+        cache.setMaxSize(0);
+        cache.save("user1", Arrays.asList("segment1", "segment2"));
+        cache.save("user2", Arrays.asList("segment3", "segment4"));
+        cache.save("user3", Arrays.asList("segment5", "segment6"));
+
+        assertNull(cache.lookup("user1"));
+        assertNull(cache.lookup("user2"));
+        assertNull(cache.lookup("user3"));
+    }
+
+    @Test
+    public void whenItemsExpire() throws InterruptedException {
+        DefaultLRUCache<List<String>> cache = new DefaultLRUCache<>();
+        cache.setTimeout(1L);
+        cache.save("user1", Arrays.asList("segment1", "segment2"));
+        assertEquals(Arrays.asList("segment1", "segment2"), cache.lookup("user1"));
+        assertEquals(1, cache.linkedList.size());
+        assertEquals(1, cache.hashMap.size());
+        Thread.sleep(1000);
+        assertNull(cache.lookup("user1"));
+        assertEquals(0, cache.linkedList.size());
+        assertEquals(0, cache.hashMap.size());
+    }
+
+    @Test
+    public void whenCacheReachesMaxSize() {
+        DefaultLRUCache<List<String>> cache = new DefaultLRUCache<>();
+        cache.setMaxSize(2);
+
+        cache.save("user1", Arrays.asList("segment1", "segment2"));
+        cache.save("user2", Arrays.asList("segment3", "segment4"));
+        cache.save("user3", Arrays.asList("segment5", "segment6"));
+
+        assertEquals(2, cache.linkedList.size());
+        assertEquals(2, cache.hashMap.size());
+
+        assertEquals(Arrays.asList("segment5", "segment6"), cache.lookup("user3"));
+        assertEquals(Arrays.asList("segment3", "segment4"), cache.lookup("user2"));
+        assertNull(cache.lookup("user1"));
+    }
+
+    @Test
+    public void whenMaxSizeIsReducedInBetween() {
```

Comment:
We can remove this test

Reply:
Modified the test to check the changed implementation